plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/RetryingConnectionFactoryModule.java

```diff
@@ -0,0 +1,41 @@
+package io.trino.plugin.jdbc;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import com.google.inject.Scopes;
+import com.google.inject.Singleton;
+import dev.failsafe.RetryPolicy;
+import io.trino.plugin.jdbc.RetryingConnectionFactory.DefaultRetryStrategy;
+import io.trino.plugin.jdbc.RetryingConnectionFactory.RetryStrategy;
+import io.trino.spi.TrinoException;
+
+import static com.google.inject.multibindings.OptionalBinder.newOptionalBinder;
+import static java.time.temporal.ChronoUnit.MILLIS;
+import static java.time.temporal.ChronoUnit.SECONDS;
+
+public class RetryingConnectionFactoryModule
+        extends AbstractModule
+{
+    @Override
+    public void configure()
+    {
+        bind(RetryingConnectionFactory.class).in(Scopes.SINGLETON);
+        newOptionalBinder(binder(), RetryStrategy.class)
+                .setDefault()
+                .to(DefaultRetryStrategy.class)
+                .in(Scopes.SINGLETON);
+    }
+
+    @Provides
+    @Singleton
+    public static RetryPolicy<Object> getRetryPolicy(RetryStrategy retryStrategy)
```

Comment:
I don't think it should be part of the guice context. I would move it as implementation detail of `RetryingConnectionFactory`

Reply:
Will propose a different way of hiding the implementation detail, ie. use PrivateModule.

Reply:
I'm fine with minimalistic approaches. Even prefer them. How would you inject the strategy then? `RetryStrategy retryStrategy`

Reply:
Moved to static field